crates/bevy_remote/src/schemas/reflect_info.rs

```diff
@@ -0,0 +1,2332 @@
+            .collect();
+        AttributesInformation(Arc::new(map))
+    }
+}
+
+impl AttributeInfoReflect for AttributesInformation {
+    fn try_get_attribute_by_id(&self, id: TypeId) -> Option<&dyn Reflect> {
+        self.get(&id).map(|s| &**s)
+    }
+}
+
+impl AttributeInfoReflect for CustomAttributes {
+    fn try_get_attribute_by_id(&self, id: TypeId) -> Option<&dyn Reflect> {
+        self.get_by_id(id)
+    }
+}
+
+pub(super) trait OptionalInfoReader {
+    fn try_get_optional(&self) -> Option<&GenericInfo>;
+}
+
+impl OptionalInfoReader for EnumInfo {
+    fn try_get_optional(&self) -> Option<&GenericInfo> {
+        let generic = self.generics().first()?;
+        if self.variant_len() != 2
+            || !self.contains_variant("Some")
+            || !self.contains_variant("None")
+        {
+            return None;
+        }
+        Some(generic)
+    }
+}
+impl OptionalInfoReader for TypeInfo {
+    fn try_get_optional(&self) -> Option<&GenericInfo> {
+        let TypeInfo::Enum(enum_info) = self else {
+            return None;
+        };
+        enum_info.try_get_optional()
+    }
+}
+impl OptionalInfoReader for TypeRegistration {
+    fn try_get_optional(&self) -> Option<&GenericInfo> {
+        self.type_info().try_get_optional()
+    }
+}
+
+fn try_get_regex_for_type(id: TypeId) -> Option<Cow<'static, str>> {
+    let data = BASE_TYPES_INFO.get(&id)?;
+    match data.schema_type {
+        SchemaType::Number => Some("\\d+(?:\\.'\\d+)?".into()),
+        SchemaType::Integer => Some("^(0|-*[1-9]+[0-9]*)$".into()),
+        _ => None,
+    }
+}
+
+/// Represents the data of a field in a schema.
+#[derive(Clone, Reflect, Debug)]
+pub struct SchemaFieldData {
+    /// Name of the field.
+    pub name: Option<Cow<'static, str>>,
+    /// Index of the field. Can be provided for named fields when the data is obtained from containing struct definition.
+    pub index: Option<usize>,
+    /// Description of the field.
+    pub description: Option<Cow<'static, str>>,
+    /// Custom of the field.
+    pub range: Option<MinMaxValues>,
+    /// Type of the field.
+    pub type_id: TypeId,
+}
+
+impl SchemaFieldData {
+    /// Returns the name of the field.
+    pub fn to_name(&self) -> Cow<'static, str> {
+        match &self.name {
+            Some(name) => name.clone(),
+            None => Cow::Owned(format!("[{}]", self.index.unwrap_or(0))),
+        }
+    }
+    /// Returns the description of the field.
+    pub fn to_description(&self) -> Option<Cow<'static, str>> {
+        self.description
+            .as_ref()
+            .map(|description| Cow::Owned(description.trim().replace("\n", "")))
+    }
+}
+
+/// Stores information about the location and id of a reference in a JSON schema.
+#[derive(Debug, Clone, PartialEq, Default, Reflect, Hash, Eq, Ord, PartialOrd)]
+pub struct TypeReferencePath {
+    /// The location of the reference in the JSON schema.
+    pub localization: ReferenceLocation,
```

Comment:
`localization` or `location`?
```suggestion
    pub location: ReferenceLocation,
```

Reply:
Good call, updated